files/image_config/ntp/mgmtvrf_add_cgexec_to_ntp.sh

```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+sudo sed -i '/start-stop-daemon --start --quiet --oknodo --pidfile $PIDFILE --startas $DAEMON -- -p $PIDFILE $NTPD_OPTS/d' $1
+sudo sed -i '/flock -w 180 9/a \\t\t\tvrfEnabled=$(/usr/local/bin/sonic-cfggen -d -v '"'MGMT_VRF_CONFIG[\"vrf_global\"][\"mgmtVrfEnabled\"]'"')\n\t\t\tif [ "$vrfEnabled" = "true" ]\n\t\t\tthen\n\t\t\t\tlog_daemon_msg "Starting NTP server in mgmt-vrf" "ntpd"\n\t\t\t\tcgexec -g l3mdev:mgmt start-stop-daemon --start --quiet --oknodo --pidfile $PIDFILE --startas $DAEMON -- -p $PIDFILE $NTPD_OPTS\n\t\t\telse\n\t\t\t\tlog_daemon_msg "Starting NTP server in default-vrf" "ntpd"\n\t\t\t\tstart-stop-daemon --start --quiet --oknodo --pidfile $PIDFILE --startas $DAEMON -- -p $PIDFILE $NTPD_OPTS\n\t\t\tfi' $1
```

Comment:
it is difficult to read this file, can we just write our own ntp service file?

Reply:
We shall create ntp service file and include the checks for MGMT_VRF enable and update the PR.

Reply:
W.r.t this file /etc/init.d/ntp file, this is installed as part of the NTP package installation. This uses "start-stop-daemon" executable to start the NTP process. The above SED method is to execute this "start-stop-daemon" using "cgexec -g l3mdev:mgmt". One option to avoid the SED is to have a new service "ntp-mgmt-vrf.service". But, we need to copy this /etc/init.d/ntp file to this new service and modify it to use cgexec. Is that fine? or, shall we keep this SED method? This sed is just adding a "if" condition to check if mgmtVRF is enabled and then it uses "cgexec" to call start-stop-daemon.

Reply:
@lguohan : We think that starting a service using "cgexec" is not the right way. The actual process should be started using "cgexec". When we modify the /etc/init.d/ntp file, we are using the following command (this is where we used SED to update this file).
"cgexec -g l3mdev:mgmt start-stop-daemon --start --quiet --oknodo --pidfile $PIDFILE --startas $DAEMON -- -p $PIDFILE $NTPD_OPTS" where it starts the ntpd daemon using cgexec.
Whereas, if we create a new service "ntp-mvrf", we still need to modify the line at which the ntpd daemon is getting started. Just by calling "cgexec -g l3mdev:mgmt service ntp-mvrf start" is not solving the problem. Hence, we suggest to retain this SED. Request your feedback on this.

Reply:
In order to execute the NTP daemon in the context of management vrf using cgexec, the NTP script /etc/init.d/ntp is updated to check for mvrf enabled status and call "start-stop-daemon" using cgexec. Since this /etc/init.d/ntp script is installed as part of NTP application (apt-get), a new script is used to update the installed /etc/init.d/ntp.
Summary: Only change is to check mvrf enabled status and call start-stop-daemon using "cgexec" in the already installed file /etc/init.d/ntp.

Reply:
I think the comment here is not to use sed, but provide our own (vrf awared) ntp start file for better readability.

Reply:
@lguohan : Removed sed usage and created our own file for ntp start where the vrf enable status is checked.